tsdb/shard.go

```diff
@@ -178,23 +223,93 @@ func (s *Shard) WritePoints(points []Point) error {
 			}
 		}
 
-		// save the raw point data
+		// Write points to WAL bucket.
+		wal := tx.Bucket([]byte("wal"))
 		for _, p := range points {
-			bp, err := tx.CreateBucketIfNotExists(p.Key())
+			// Generate an autoincrementing index for the WAL.
+			id, _ := wal.NextSequence()
```

Comment:
Ah, nope, not yet. I missed that. I think I might actually change the flusher to go off the WAL bucket instead. It'll make consistency easier.